include/s3select_functions.h

```diff
@@ -1258,6 +1261,87 @@ struct _fn_like : public base_function
   }
 };
 
+struct _fn_likescape : public base_function
+{
+  void transform(std::string& s, size_t found)
+  {
+    std::string::size_type i = 0;
+    while (!s.empty())
+    {
+        i = s.find("%%", i);
+        if (i == std::string::npos) break;
+        s.erase(i, 1);      
+    }
+    bool startswith = (s[0] == '%' && s[s.size()-1] != '%');
+    bool endswith = (s[0] != '%' && s[s.size()-1] == '%');
+    bool startend = (s[0] == '%' && s[s.size()-1] == '%');
+    if (startswith) {
+      std::replace(s.begin(), s.begin()+1, '%', '^');
+      s.insert(1, ".");
+      s.insert(2, "*");
+      s.append("$");
+      std::replace(s.begin(), s.begin()+found+2, '_', '.');
+      std::replace(s.begin()+found+3, s.end(), '_', '.');
+    } else if (endswith) {
+      s.insert(0, "^");
+      std::replace(s.end()-1, s.end(), '%', '$');
+      s.insert(s.size()-1,".");
+      s.insert(s.size()-1,"*");
+      std::replace(s.begin(), s.begin()+found+1, '_', '.');
+      std::replace(s.begin()+found+2, s.end(), '_', '.');
+    } else if (startend) {
+      std::replace(s.begin(), s.begin()+1, '%', '^');
+      s.insert(1, ".");
+      s.insert(2, "*");
+      std::replace(s.end()-1, s.end(), '%', '$');
+      s.insert(s.size()-1,".");
+      s.insert(s.size()-1,"*");
+      std::replace(s.begin(), s.begin()+found+2, '_', '.');
+      std::replace(s.begin()+found+3, s.end(), '_', '.');
+    } else {
+      std::string findThis = "%";
+      std::string replaceWith = ".*";
+      s.insert(0,"^");
+      s.append("$");
+      std::size_t pos = 0;
+      while ((pos = s.find(findThis, pos)) != std::string::npos)
+      {
+        s.replace(pos, findThis.length(), replaceWith);
+        pos += replaceWith.length();
+      }
+      std::replace(s.begin(), s.begin()+found+1, '_', '.');
+      std::replace(s.begin()+found+2, s.end(), '_', '.');
+    }
+  }
+  
+  bool operator()(bs_stmt_vec_t* args, variable* result) override
+  {
+    auto iter = args->begin();
+    base_statement* escape_expr = *iter;
+    iter++;
+    base_statement* like_expr = *iter;
+    iter++;
+    base_statement* main_expr = *iter;
+
+    value escape_expr_val = escape_expr->eval();
+    value like_expr_val = like_expr->eval();
+    value main_expr_val = main_expr->eval();
+
+    std::string like_string = like_expr_val.to_string();
+    std::string escape_char = escape_expr_val.to_string();
+    size_t found = like_string.find(escape_char[0]);
+    like_string.erase (std::remove(like_string.begin(), like_string.end(), escape_char[0]), like_string.end());
+    transform(like_string, found);
+    std::regex compiled_regex = std::regex(like_string);
```

Comment:
please verify with AWS. whether the /like/ string could dynamic or , it should be constant string.

it will impact both operators like and escape_like
the impact is on performance.